src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -1537,5 +1537,11 @@ internal enum ErrorCode
         ERR_EscapeStackAlloc = 8353,
         ERR_RefReturnThis = 8354,
         #endregion diagnostics introduced for `ref readonly`, `ref ternary` and `ref-like` features in C# 7.2
+
+        #region diagnostics introduced for recursive patterns
+        // PROTOTYPE(patterns2): renumber these before committing
+        ERR_MissingPattern = 8500,
```

Comment:
>8500 [](start = 29, length = 4)

It looks like the 8500 range is already used by Default Interface Implementations

Reply:
Sounds like we need a global registry.

Reply:
Recorded in #23222.